lib/asn1.ml

```diff
@@ -478,23 +480,21 @@ module Algorithm_identifier = struct
       | Rsa
       | Ec
       | Dh
-      | Unknown of Asn.OID.t
 
     let grammar =
       let open Asn.S in
       let f = function
         | oid when oid = rsa_oid -> Rsa
-        | oid when oid = dsa_oid -> Dsa
+        | oid when oid = dsa_oid || oid = dsa_oiw -> Dsa
         | oid when oid = ec_oid || oid = ec_dh || oid = ec_mqv -> Ec
         | oid when oid = dh_oid -> Dh
-        | oid -> Unknown oid
```

Comment:
Before this fix, the combinator would continue with the `Unknown` constructor, most of the time raising the error later. Now it raises it here. Are we sure that's what we want ?

Reply:
I see your point, although as far as I can tell, these `Algo.t` are only ever consumed by the `dsa_grammar`, `rsa_grammar`, `dh_grammar` and `ec_grammar` functions inside the `Algorithm_identifier` module, in which case it makes little difference. But your `to_string` idea is cute, and easier than trying to create a general non-crashing `parse_warning` feature, which was what I had in mind.